### ISSUE TITLE
Unexpected method 'appcast' called on Cask winds

### ISSUE THREAD

**Issue by jsingal** [Homebrew/homebrew-cask#173201]
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match / SHA256 mismatch](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error (do not open an issue before trying to open a PR to fix first).

### Description of issue

Encounter this error when running 'brew upgrade' or 'brew upgrade --cask' command.

Unexpected method 'appcast' called on Cask winds

### Command that failed

brew upgrade

### Output of command with `--verbose --debug`

```shell
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/testssl
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/tree
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libheif
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libtiff
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libtool
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/gmp
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/wget
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libidn2
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromTapLoader): loading teamookla/speedtest/speedtest
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/little-cms2
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libpng
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/ncurses
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/ghostscript
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/nettle
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/pixman
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/gdbm
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/mpdecimal
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/freetype
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/imath
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/liblinear
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/starship
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libunistring
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/liblqr
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/openjpeg
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/highway
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/ldns
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/brotli
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libslirp
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/shared-mime-info
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/concurrencykit
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/python@3.12
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/openssh
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/glib
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/readline
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/aom
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libde265
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/lz4
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/darksky-weather
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/webp
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/sqlite
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/xz
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libnghttp2
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/vde
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/lua
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/openexr
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/fontconfig
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromTapLoader): loading homebrew/core/archey
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromKegLoader): loading archey
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/htop
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/giflib
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/certifi
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/capstone
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libfido2
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/dtc
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/snappy
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libvmaf
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/openssl@3
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/ca-certificates
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/gettext
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libssh2
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libcbor
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/qemu
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/zstd
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/ansiweather
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/tcl-tk
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/dnsperf
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/jbig2dec
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/m4
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libtasn1
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/oniguruma
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libraw
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libssh
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/pcre2
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/youtube-dl
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libevent
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/openssl@1.1
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/pcre
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/jpeg-xl
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/python@3.10
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libidn
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/x265
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libomp
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/unbound
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/nmap
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/jq
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/zsh-syntax-highlighting
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libusb
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/python@3.11
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/imagemagick
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/shodan
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/jasper
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/p11-kit
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/lzo
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/libffi
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/telnet
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/python@3.9
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/jpeg-turbo
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/gnutls
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromAPILoader): loading homebrew/core/smap
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading 1password
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading 1password-cli
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading adobe-acrobat-reader
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading aerial
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading airparrot
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading appcleaner
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading atom
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading authy
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading bartender
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading brave-browser
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading cakebrew
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading cardhop
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading clean-me
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading cloudflare-warp
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading cyberduck
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading docker
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading duet
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromInstalledPathLoader): loading exodus
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading fantastical
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading firefox
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading google-chrome
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading google-drive
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading google-earth-pro
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading handbrake
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading hazel
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading hyper
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading iina
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading istat-menus
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading iterm2
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading itsycal
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading keka
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading kindle
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading little-snitch
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading microsoft-auto-update
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading microsoft-office
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading microsoft-teams
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading mimestream
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading mountain-duck
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading mullvadvpn
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading numi
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading pdf-expert
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading popclip
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading protonvpn
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading pulsar
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading rocket-typist
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading shottr
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading spotify
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading steam
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading sublime-text
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromInstalledPathLoader): loading subnetcalc
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading tg-pro
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading the-unarchiver
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading thunderbird
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading tor-browser
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading transmission
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading transmit
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading utm
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading visual-studio-code
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading vlc
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading whatsapp
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading wifi-explorer
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromInstalledPathLoader): loading winds
Error: Unexpected method 'appcast' called on Cask winds.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading zoom
```

### Output of `brew doctor` and `brew config`

```shell
➜  ~ brew doctor
Error: Unexpected method 'appcast' called on Cask winds.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Error: Unexpected method 'appcast' called on Cask winds.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  atom
  authy
  cakebrew
  clean-me
  kindle

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  archey

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  darksky-weather
  openssl@1.1
  youtube-dl

Warning: Some installed formulae are not readable:
  archey: wrong number of arguments (given 1, expected 0)


➜  ~ brew config
HOMEBREW_VERSION: 4.2.21-47-g9bf7734
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9bf7734749a0d8e99d2e3f966bb236a08f2a2609
Last commit: 54 minutes ago
Core tap HEAD: 88c922d97a51710cbd36788e2c1b9d4e1f1eb520
Core tap last commit: 30 minutes ago
Core tap JSON: 08 May 12:14 UTC
Core cask tap HEAD: 143403b45228fc215ac9c14efc22de0ec2345573
Core cask tap last commit: 48 minutes ago
Core cask tap JSON: 08 May 12:14 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew tap`

```shell
➜  ~ brew tap
1password/tap
homebrew/bundle
homebrew/cask
homebrew/core
homebrew/services
teamookla/speedtest
wpscanteam/tap
```

**Comment by donpark** [comment]
Some observations and temporary workaround:

- `brew upgrade` seems to work fine despite the error.
- removing module blamed in the error resolved the error message.

Error:

```bash
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromInstalledPathLoader): loading minikube
Error: Unexpected method 'appcast' called on Cask minikube.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
```

Temporary Workaround:

```bash
brew uninstall minikube
```

**Comment by xand3r40r93** [comment]
Error: Unexpected method 'appcast' called on Cask vimmotion.

**Comment by bevanjkay** [comment]
In all of these cases, this is caused by having a Cask still installed in your Caskroom that was removed from `homebrew-cask` more than a couple of years ago. They were therefore never updated to remove the deprecated `appcast` stanza.
If you wish to keep the app installed, you may be best off removing the Cask from your Caskroom (however note that you won't be able to uninstall the app using Homebrew), as it will never receive updates or be able to be reinstalled in future.

`rm -rfi $(brew --caskroom)/<token>`

`rm -rfi $(brew --caskroom)/winds`

This will not remove the application from your computer.

**Comment by jsingal** [comment]
Thanks for your guys help.

**Comment by lodewykk** [comment]
I received the same issue with adoptopenjdk:

`brew uninstall ffmpeg`
``` sh 
Error: Unexpected method 'appcast' called on Cask adoptopenjdk15.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Error: Unexpected method 'appcast' called on Cask adoptopenjdk8.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
```

Removing the caskroom entries using the instructions from @bevanjkay solved the issue.

**Comment by sgup** [comment]
Unexpected method 'appcast' called on Cask adoptopenjdk11.

**Comment by gadgetvirtuoso** [comment]
> Unexpected method 'appcast' called on Cask adoptopenjdk11.
Had the same. I ended up deleting the adoptopenjdk manually then reinstall if needed, but this one is being deprecated for temurin.
`/opt/homebrew/Caskroom/adoptopenjdk`

**Comment by syntax76** [comment]
I got the same problem for tcl:

```Unexpected method 'appcast' called on Cask tcl.```

**Comment by Twoody** [comment]
I had a similar issue with `adoptopenjdk8`, despite not having the cask installed, and going through the loops with `temurin`.

After removing the cask manually, having no errors with `brew doctor`, but I still had errors with `brew cleanup`.

**Anyway, to solve the error:** 

1. `brew edit --cask adoptopenjdk8`
2. Remove or comment out the line containing `appcast`.
3. `brew install --cask adoptopenjdk8 `

**Comment by MiniSC** [comment]
thank for this suggest, i use brew --caskroom to find my caskroom position then remove it , that's worked

> In all of these cases, this is caused by having a Cask still installed in your Caskroom that was removed from `homebrew-cask` more than a couple of years ago. They were therefore never updated to remove the deprecated `appcast` stanza. If you wish to keep the app installed, you may be best off removing the Cask from your Caskroom (however note that you won't be able to uninstall the app using Homebrew), as it will never receive updates or be able to be reinstalled in future.
> 
> `rm -rfi $(brew --caskroom)/<token>`
> 
> `rm -rfi $(brew --caskroom)/winds`
> 
> This will not remove the application from your computer.

**Comment by jameswomack** [comment]
This worked for me

```sh
/usr/bin/find "$(brew --prefix)/Caskroom/"*'/.metadata' -type f -name '*.rb' -print0 | /usr/bin/xargs -0 /usr/bin/perl -i -pe 's/depends_on macos: \[.*?\]//gsm;s/depends_on macos: .*//g'
```

[Source](https://shaunstanislaus.medium.com/fixes-for-big-surs-homebrew-69137f848eb8)

**Comment by gk258007** [comment]
Looked up for the --caskroom dir from home-brew and removed it. Thank you @bevanjkay ! Cheers!

**Comment by Elmago34** [comment]
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

**Comment by Chanzhaoyu** [comment]
> I had a similar issue with `adoptopenjdk8`, despite not having the cask installed, and going through the loops with `temurin`.I had a similar issue with `adoptopenjdk8`, despite not having the cask installed, and going through the loops with `temurin`.尽管没有安装木桶，并且使用`temurin`进行循环，但我在`adoptopenjdk8`时也遇到了类似的问题。
> 
> After removing the cask manually, having no errors with `brew doctor`, but I still had errors with `brew cleanup`.After removing the cask manually, having no errors with `brew doctor`, but I still had errors with `brew cleanup`.手动移除木桶后，`brew doctor`没有错误，但我仍然在`brew cleanup`中出现错误。
> 
> **Anyway, to solve the error:Anyway, to solve the error: 无论如何，要解决错误：**
> 
> 1. `brew edit --cask adoptopenjdk8`
> 2. Remove or comment out the line containing `appcast`.Remove or comment out the line containing `appcast`.删除或注释掉包含`appcast`的行。
> 3. `brew install --cask adoptopenjdk8 `

Thanks

**Comment by alienlifeform** [comment]
This also worked for me. 

> `brew edit --cask adoptopenjdk8`
> # Remove or comment out the line containing appcast.
> `brew install --cask adoptopenjdk8`

**Comment by anuvedantham** [comment]
For anyone still encountering this error on macOS, the package's current [installation instructions](https://github.com/PyMySQL/mysqlclient/blob/3805b28abc24cc985ffe7c8a05dd4bd6530aefe8/README.md#macos-homebrew) worked for me:
```
$ brew install mysql pkg-config
$ pip install mysqlclient
```

**Comment by sqmw** [comment]
So abstract a problem, I also encountered

**Comment by sdaitzman** [comment]
Leaving a comment now for anyone else who encounters the text "Error: Unexpected method 'appcast' called on Cask now."

`now` is the name of the Cask, not part of the message! You need to remove the `now` package that uses the deprecated entry to resolve the error message.

**Comment by watsaket00** [comment]
> In all of these cases, this is caused by having a Cask still installed in your Caskroom that was removed from `homebrew-cask` more than a couple of years ago. They were therefore never updated to remove the deprecated `appcast` stanza. If you wish to keep the app installed, you may be best off removing the Cask from your Caskroom (however note that you won't be able to uninstall the app using Homebrew), as it will never receive updates or be able to be reinstalled in future.
> 
> `rm -rfi $(brew --caskroom)/<token>`
> 
> `rm -rfi $(brew --caskroom)/winds`
> 
> This will not remove the application from your computer.

This work for me too, Thank you :)

**Comment by Edjevw12** [comment]
`brew cleanup --prune=all`

this worked for me

**Comment by jobearrr** [comment]
I only needed to comment out the line containing `appcast` as suggested here.
> 1. `brew edit --cask adoptopenjdk8`
> 2. Remove or comment out the line containing `appcast`.
> 3. `brew install --cask adoptopenjdk8 `

**Comment by dennismphil** [comment]
> Error: Unexpected method 'appcast' called on Cask adoptopenjdk11.

   
The solution quoted worked with a slight tweak, the version was  `` `adoptopenjdk11` `` which I did not realize in the first glance. In case this helps anyone. 

> **Anyway, to solve the error:**
> 
> 1.  `brew edit --cask adoptopenjdk8`
> 2.  Remove or comment out the line containing `appcast`.
> 3.  `brew install --cask adoptopenjdk8`

**Comment by khuddite** [comment]
> 1. brew edit --cask adoptopenjdk8

Sure, worked for me. Thanks!

**Comment by abhi-dpe** [comment]
> `brew cleanup --prune=all`

This worked for me.

**Comment by renniek** [comment]
> `rm -rfi $(brew --caskroom)/<token>`



> In all of these cases, this is caused by having a Cask still installed in your Caskroom that was removed from `homebrew-cask` more than a couple of years ago. They were therefore never updated to remove the deprecated `appcast` stanza. If you wish to keep the app installed, you may be best off removing the Cask from your Caskroom (however note that you won't be able to uninstall the app using Homebrew), as it will never receive updates or be able to be reinstalled in future.
> 
> `rm -rfi $(brew --caskroom)/<token>`
> 
> `rm -rfi $(brew --caskroom)/winds`
> 
> This will not remove the application from your computer.

Hi there, I tried to follow these steps to resolve this error, but I am getting the error: -bash: syntax error near unexpected token `newline'. I am working on a macOS 11.1. Thanks so much!

**Comment by soloradish** [comment]
> `rm -rfi $(brew --caskroom)/<token>`
> 
> `rm -rfi $(brew --caskroom)/winds`

You should double quote the path.
```
rm -rfi "$(brew --caskroom)/<token>"
rm -rfi "$(brew --caskroom)/winds"
```

**Comment by syukronarie** [comment]
> /opt/homebrew/Caskroom/adoptopenjdk

thanks, it saves my time!

**Comment by lukasT** [comment]
for me sadly none of the commands worked so I had to:

1. `brew --caskroom`
2. `rm -rf ${replace the path with whatever the above returns}/docker-toolbox` 

(as in my case, I had problems with docker-toolbox but you can of course replace it with the library causing trouble for you)
